{% extends 'main/lesson_base.html' %} {% load static %} {% block html_of_lesson %}
<div class="main-text">
    <h4>Ерекшелік дегеніміз не?</h4>
    <p>Сіз <u>Мәтіндік редактор</u> бағдарламасын әзірледіңіз делік. Бағдарламада пайдаланушы жаңа файл жасай алады, оған деректерді енгізеді, содан кейін файлды жүйеде сақтай алады.</p>
    <p>
    <br>
    </p>
    <p>Егер код дұрыс жазылса, қателер болмайды. Бірақ пайдаланушы редакторды ашып, қажетті файлды ашты, оған деректерді жазды, содан кейін файлды компьютерден қолмен жойды, содан кейін файлды сіздің бағдарламаңыз арқылы сақтауға тырысты деп елестетіп көрейік.&nbsp;</p>
    <p>
    <br>
    </p>
    <p>Бұл сценарийде сіз бағдарламаны бұзатын және пайдаланушының бағдарламаңызда жұмыс істеуге деген ықыласын кетіретін қатені аласыз.&nbsp;</p>
    <p>
    <br>
    </p>
    <p>Ерекшелік бағдарламаның өзі жұмыс істеген кезде пайда болатын қате болып табылады. Шартты мәлімдемелер арқылы мұндай қателерді қадағалау әрқашан мүмкін бола бермейді, өйткені бағдарлама қазірдің өзінде жұмыс істеп тұр, біз пайдаланушының әрекетін алдын ала білмейміз, яғни оны жасау кезінде қатені «ұстап алу» керек.</p>
    <p>
    <br>
    </p>
    <h4>Қателерді ұстау</h4>
    <p>Жоғарыдағы жағдай үшін қателерді жасалған сәтте анықтау үшін try except құрылымы қажет.&nbsp;</p>
    <p>
    <br>
    </p>
    <p>Бұл құрылымның көмегімен әртүрлі пішімдер мен оқиғалардың қателерін бақылай аласыз. Сіз файлдың қате ашылуын бақылай аласыз, сыныптың қате қосылымын қадағалай аласыз, айнымалының немесе функцияның қате жазылуын қадағалай аласыз және онда не бар, тіпті сандарды нөлге бөлуді бақылай аласыз!</ p>
    <p>
    <br>
    </p>
    <h4> try except construct </h4> іске асырылуы
    <p>Қатені бақылауды қосу үшін келесі кодты жазуға болады:</p><pre class=" language-python"><code class="lang  language-python"><span class="token keyword">try</span><span class="token punctuation">:</span> <span class="token comment"># Пробуем что-то сделать</span>
	k <span class="token operator">=</span> <span class="token number">1</span> <span class="token operator">/</span> <span class="token number">0</span> <span class="token comment"># Деление на ноль</span>
<span class="token keyword">except</span> ArithmeticError<span class="token punctuation">:</span> <span class="token comment"># Отслеживаем на арифметическую ошибку</span>
	<span class="token keyword">print</span> <span class="token punctuation">(</span><span class="token string">"Найдена ошибка"</span><span class="token punctuation">)</span> <span class="token comment"># Мы нашли ошибку</span>
	k <span class="token operator">=</span> <span class="token number">0</span> <span class="token comment"># Наша переменная теперь будет равна 0</span>

<span class="token keyword">print</span><span class="token punctuation">(</span>k<span class="token punctuation">)</span> <span class="token comment"># Просто выводим переменную</span></code></pre>
<p>Жоғарыдағы код ерекше жағдайларды пайдаланудың өте қарапайым мысалын көрсетеді. Оның қарапайымдылығын елемеңіз, өйткені ең бастысы try except block-ның жалпы құрылысын қарастыру.</p>
<p>
<br>
</p>
<p>Бұл конструкцияны пайдаланған кезде, бағдарлама барысында қандай қате орын алса да, бағдарламаның өзі жұмысын тоқтатпайды және пайдаланушы оны пайдалануда жағымсыз тәжірибе алмайды.</p>
<p>
<br>
</p>
<p>Егер бірнеше қате класын қадағалау қажет болса, блоктардан басқа бірнеше қосуға болады. Олардың барлығы шартты операторлардағы сияқты бірінің астына қосылады.&nbsp;</p>
</div> {% endblock %}